cmd/watch-termination/main.go

```diff
@@ -53,6 +65,54 @@ func main() {
 		klog.SetOutputBySeverity("INFO", stderr)
 	}
 
+	var client kubernetes.Interface
+	if len(*kubeconfigPath) > 0 {
+		loader := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{ExplicitPath: *kubeconfigPath}, nil)
+		if cfg, err := loader.ClientConfig(); err != nil {
+			klog.Errorf("failed to load kubeconfig %q: %v", *kubeconfigPath, err)
+			return 1
+		} else {
+			client = kubernetes.NewForConfigOrDie(cfg)
+		}
+	}
+
+	// touch file early. If the file is not removed on termination, we are not
+	// terminating cleanly via SIGTERM.
+	if len(*terminationLock) > 0 {
+		ref, err := eventReference()
+		if err != nil {
+			klog.Errorf("failed to get event target: %v", err)
+		}
+
+		if st, err := os.Stat(*terminationLock); err == nil {
+			klog.Warningf("Previous pod did not terminate gracefully: %s", st.ModTime().String())
+			if client != nil {
+				go wait.PollUntil(5*time.Second, func() (bool, error) {
```

Comment:
i would retry longer

Reply:
it retries infinitely, but only every 5s.